yarn-project/pxe/src/config/index.ts

```diff
@@ -34,44 +35,79 @@ export interface PXEConfig {
+  l2BlockPollingIntervalMS: {
+    env: 'PXE_BLOCK_POLLING_INTERVAL_MS',
+    description: 'The interval to wait between polling for new blocks.',
+    ...numberConfigHelper(1_000),
+  },
+  l2StartingBlock: {
+    env: 'PXE_L2_STARTING_BLOCK',
+    ...numberConfigHelper(INITIAL_L2_BLOCK_NUM),
+    description: 'L2 block to start scanning from for new accounts',
+  },
+  dataDirectory: {
+    env: 'PXE_DATA_DIRECTORY',
+    description: 'Where to store PXE data. If not set, will store in memory',
+  },
+  bbBinaryPath: {
+    env: 'BB_BINARY_PATH',
+    description: 'Path to the BB binary',
+  },
+  bbWorkingDirectory: {
+    env: 'BB_WORKING_DIRECTORY',
+    description: 'Working directory for the BB binary',
+  },
+  proverEnabled: {
+    env: 'PXE_PROVER_ENABLED',
+    parseEnv: (val: string) => ['1', 'true'].includes(val),
+    description: 'Enable real proofs',
+    isBoolean: true,
+  },
+};
 
 /**
  * Creates an instance of PXEServiceConfig out of environment variables using sensible defaults for integration testing if not set.
  */
 export function getPXEServiceConfig(): PXEServiceConfig {
-  const {
-    PXE_BLOCK_POLLING_INTERVAL_MS,
-    PXE_L2_STARTING_BLOCK,
-    PXE_DATA_DIRECTORY,
-    BB_BINARY_PATH,
-    BB_WORKING_DIRECTORY,
-    PXE_PROVER_ENABLED,
-  } = process.env;
-
-  return {
-    l2BlockPollingIntervalMS: PXE_BLOCK_POLLING_INTERVAL_MS ? +PXE_BLOCK_POLLING_INTERVAL_MS : 1000,
-    l2StartingBlock: PXE_L2_STARTING_BLOCK ? +PXE_L2_STARTING_BLOCK : INITIAL_L2_BLOCK_NUM,
-    dataDirectory: PXE_DATA_DIRECTORY,
-    bbBinaryPath: BB_BINARY_PATH,
-    bbWorkingDirectory: BB_WORKING_DIRECTORY,
-    proverEnabled: ['1', 'true'].includes(PXE_PROVER_ENABLED!),
-  };
+  return getConfigFromMappings<PXEServiceConfig>(pxeConfigMappings);
 }
 
+export const pxeCliConfigMappings: ConfigMappingsType<CliPXEOptions> = {
+  network: {
+    env: 'NETWORK',
+    parseEnv: (val: string) => val as Network,
+    description: 'External Aztec network to connect to. e.g. devnet',
+  },
+  apiKey: {
+    env: 'API_KEY',
+    description: "API Key required by the external network's node",
+  },
+  nodeUrl: {
+    env: 'AZTEC_NODE_URL',
+    description: 'Custom Aztec Node URL to connect to',
+  },
+};
+
 /**
  * Creates an instance of CliPxeOptions out of environment variables
  */
-export function getCliPXEOptions(): CliPXEOptions {
-  const pxeServiceConfig = getPXEServiceConfig();
-  const { NETWORK, AZTEC_NODE_URL, ΑPI_KEY } = process.env;
-
+export function getCliPXEOptions(): CliPXEOptions & PXEServiceConfig {
+  const pxeConfig = getPXEServiceConfig();
+  const cliOptions = getConfigFromMappings<CliPXEOptions>(pxeCliConfigMappings);
   return {
-    ...pxeServiceConfig,
-    network: NETWORK ? NETWORK as Network : undefined,
-    apiKey: ΑPI_KEY,
-    nodeUrl: AZTEC_NODE_URL,
-    proverEnabled: !!NETWORK,
+    ...pxeConfig,
+    ...cliOptions,
+    proverEnabled: pxeConfig.proverEnabled || !!cliOptions.network,
```

Comment:
I know this was from before, but why is prover enabled determined by network being present?

Reply:
My understanding is we basically are going to have all our networks with proving enabled? @PhilWindle

Reply:
That's probably true. But I think the network collection of key/values should explicitly specify the value of `proverEnabled`.